tokio/src/sync/broadcast.rs

```diff
@@ -1104,8 +1107,12 @@ impl<T> Receiver<T> {
                                     }
                                 }
 
-                                if !(*ptr).queued {
-                                    (*ptr).queued = true;
+                                // If the waiter is not already queued, enqueue it.
+                                // Relaxed memory order suffices because we don't need
+                                // to synchronize with `Recv::drop` here (calling
+                                // `Receiver::recv_ref` with a waiter implies ownership
+                                // of the corresponding `Recv`).
+                                if !(*ptr).queued.swap(true, Relaxed) {
```

Comment:
Here you have exclusive access, so just perform a normal write.

Reply:
It turned out that loom `AtomicBool` does not support `get_mut`, so I resolved to using relaxed load/store.